rust_src/src/lists.rs

```diff
@@ -260,7 +263,7 @@ impl Iterator for CarIter {
 
 impl From<LispObject> for LispCons {
     fn from(o: LispObject) -> Self {
-        o.as_cons().unwrap_or_else(|| wrong_type!(Qconsp, o))
+        o.as_cons_or_error()
```

Comment:
As far as I understand (I'm not fully catching up :disappointed_relieved:), it seems to be better to use a former implementation as currently @nickdrozd and others working on improving the type conversion mechanism for Rust codes (e.g. `.as_<type>()` and `.as_<type>_or_error()` should be `into()`, inherited from `From`. We should make it as documents (#1253)). #1221 and related issues might be helpful reference.

Reply:
Good point, I wasn't aware of this!
I changed some conversions to use `into()` (or `Foo::from()` if type annotations are necessary)